gem/backends.py

```diff
@@ -39,23 +39,36 @@ def _update_user_from_claims(user, claims):
     # If the user doesn't have a profile for some reason make one
     if not hasattr(user, 'profile'):
         user.profile = UserProfile(user=user)
+
+        # TODO: we should be using a more specific site here?
         user.profile.site = Site.objects.get(is_default_site=True)
 
     # Ensure the profile is linked to their auth service account using the uuid
     if user.profile.auth_service_uuid is None:
         user.profile.auth_service_uuid = claims.get("sub")
 
+        # If a user already exists with this username
+        # change that user's username
+        username = claims.get("username")
+        if username:
+            for user in User.objects.filter(username=username):
+                user.username = user.profile.site.pk + '_' + username
```

Comment:
I think we only want to change the user name if the users that are returned are not the current user.
Also, was the decision to just add the site id to the username? Is that guaranteed to be unique or is there some case where it might clash as well?